### ISSUE TITLE
build error with sumo 1.1.0 and old ffmpeg version

### ISSUE THREAD

**Issue by mawright** [eclipse/sumo#4995]
I am building 1.1.0 from source in Ubuntu 16.04 using cmake. The compiler throws errors when it hits the file GuiViewTraffic.cpp, saying that some functions are not defined.

```
[ 48%] Building CXX object src/gui/CMakeFiles/gui.dir/GUIViewTraffic.cpp.o
In file included from /usr/local/sumo-1.1.0/src/gui/GUIViewTraffic.cpp:29:0:
/usr/local/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h: In constructor ‘GUIVideoEncoder::GUIVideoEncoder(const char*, int, int, double)’:
/usr/local/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:161:33: error: ‘av_packet_alloc’ was not declared in this scope
         myPkt = av_packet_alloc();
                                 ^
/usr/local/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h: In destructor ‘GUIVideoEncoder::~GUIVideoEncoder()’:
/usr/local/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:172:51: error: ‘avcodec_send_frame’ was not declared in this scope
         if (avcodec_send_frame(myCodecCtx, nullptr) < 0) {
                                                   ^
/usr/local/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:177:59: error: ‘avcodec_receive_packet’ was not declared in this scope
             ret = avcodec_receive_packet(myCodecCtx, myPkt);
                                                           ^
/usr/local/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:195:30: error: ‘av_packet_free’ was not declared in this scope
         av_packet_free(&myPkt);
                              ^
/usr/local/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h: In member function ‘void GUIVideoEncoder::writeFrame(uint8_t*)’:
/usr/local/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:208:55: error: ‘avcodec_send_frame’ was not declared in this scope
         int r = avcodec_send_frame(myCodecCtx, myFrame);
                                                       ^
/usr/local/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:216:59: error: ‘avcodec_receive_packet’ was not declared in this scope
             ret = avcodec_receive_packet(myCodecCtx, myPkt);
                                                           ^
src/gui/CMakeFiles/gui.dir/build.make:230: recipe for target 'src/gui/CMakeFiles/gui.dir/GUIViewTraffic.cpp.o' failed
make[2]: *** [src/gui/CMakeFiles/gui.dir/GUIViewTraffic.cpp.o] Error 1
CMakeFiles/Makefile2:4521: recipe for target 'src/gui/CMakeFiles/gui.dir/all' failed
make[1]: *** [src/gui/CMakeFiles/gui.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

This is the output of cmake:
```
-- CMAKE_BINARY_DIR: /usr/local/sumo-1.1.0/build/cmake-build
-- CMAKE_SOURCE_DIR: /usr/local/sumo-1.1.0
-- 
-- Platform: 
--     Host: Linux-4.4.0-141-generic x86_64
--     Target: Linux-4.4.0-141-generic x86_64
--     CMake: 3.5.1
--     CMake generator: Unix Makefiles
--     CMake build tool: /usr/bin/make
--     Compiler: GNU 5.4.0
-- 
-- Found Python: /usr/bin/python
-- Could NOT find GTest (missing:  GTEST_LIBRARY GTEST_INCLUDE_DIR GTEST_MAIN_LIBRARY) 
-- Found Proj: /usr/lib/x86_64-linux-gnu/libproj.so
-- Found Fox: -L/usr/lib/x86_64-linux-gnu -lFOX-1.6 -lX11 -lXext -lfreetype -lXft -lXcursor -lXrender -lXfixes -lfreetype -lfontconfig -lfreetype -lGL -lGLU -ldl -lpthread -lrt -ljpeg -lpng -ltiff -lz -lGLU -lGL
-- Could NOT find OpenSceneGraph: Found unsuitable version "3.2.1", but required is at least "3.4.0" (found /usr/lib/x86_64-linux-gnu/libosgGA.so;/usr/lib/x86_64-linux-gnu/libosgViewer.so;/usr/lib/x86_64-linux-gnu/libosgUtil.so;/usr/lib/x86_64-linux-gnu/libosgDB.so;/usr/lib/x86_64-linux-gnu/libosgText.so;/usr/lib/x86_64-linux-gnu/libosg.so;/usr/lib/x86_64-linux-gnu/libOpenThreads.so)
-- Could NOT find GL2PS (missing:  GL2PS_LIBRARY GL2PS_INCLUDE_DIR) 
-- Enabled features: Linux-4.4.0-141-generic x86_64 GNU 5.4.0 Release Proj GUI GDAL FFmpeg SWIG
-- Configuring done
-- Generating done
-- Build files have been written to: /usr/local/sumo-1.1.0/build/cmake-build
```

Let me know what other info about my system I need to provide.

**Comment by behrisch** [comment]
Your ffmpeg is too old to work with this sumo version and unfortunately our build script does not support version checking for ffmpeg. Until we added it you can either update your ffmpeg to some version greater than 3.0 or delete the ffmpeg related entries from the CMakeCache.txt or delete the devel packages for ffmpeg if you do not need them for another project.

**Comment by mawright** [comment]
Thanks for the info. I guess I should add for the record and benefit of other people that are confused by this error that this old ffmpeg version is the one provided by the main package for Ubuntu 16.04 LTS: https://packages.ubuntu.com/xenial-updates/ffmpeg

**Comment by xiaohonghua0110** [comment]
I have installed ffmpeg 3.4.4, but the problem is not solved yet.
ffmpeg version 3.4.4-1~16.04.york0 Copyright (c) 2000-2018 the FFmpeg developers
  built with gcc 5.4.0 (Ubuntu 5.4.0-6ubuntu1~16.04.10) 20160609
  configuration: --prefix=/usr --extra-version='1~16.04.york0' --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --enable-gpl --disable-stripping --enable-avresample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librubberband --enable-librsvg --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-omx --enable-openal --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libopencv --enable-libx264 --enable-shared
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libavresample   3.  7.  0 /  3.  7.  0
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100
Hyper fast Audio and Video encoder
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...


### Same issue: 
[ 47%] Building CXX object src/gui/CMakeFiles/gui.dir/GUIViewTraffic.cpp.o
In file included from /home/xiaohonghua/Documents/sumo-1.1.0/src/gui/GUIViewTraffic.cpp:29:0:
/home/xiaohonghua/Documents/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h: In constructor ‘GUIVideoEncoder::GUIVideoEncoder(const char*, int, int, double)’:
/home/xiaohonghua/Documents/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:161:33: error: ‘av_packet_alloc’ was not declared in this scope
         myPkt = av_packet_alloc();
                                 ^
/home/xiaohonghua/Documents/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h: In destructor ‘GUIVideoEncoder::~GUIVideoEncoder()’:
/home/xiaohonghua/Documents/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:172:51: error: ‘avcodec_send_frame’ was not declared in this scope
         if (avcodec_send_frame(myCodecCtx, nullptr) < 0) {
                                                   ^
/home/xiaohonghua/Documents/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:177:59: error: ‘avcodec_receive_packet’ was not declared in this scope
             ret = avcodec_receive_packet(myCodecCtx, myPkt);
                                                           ^
/home/xiaohonghua/Documents/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:195:30: error: ‘av_packet_free’ was not declared in this scope
         av_packet_free(&myPkt);
                              ^
/home/xiaohonghua/Documents/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h: In member function ‘void GUIVideoEncoder::writeFrame(uint8_t*)’:
/home/xiaohonghua/Documents/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:208:55: error: ‘avcodec_send_frame’ was not declared in this scope
         int r = avcodec_send_frame(myCodecCtx, myFrame);
                                                       ^
/home/xiaohonghua/Documents/sumo-1.1.0/src/utils/gui/div/GUIVideoEncoder.h:216:59: error: ‘avcodec_receive_packet’ was not declared in this scope
             ret = avcodec_receive_packet(myCodecCtx, myPkt);
                                                           ^
src/gui/CMakeFiles/gui.dir/build.make:230: recipe for target 'src/gui/CMakeFiles/gui.dir/GUIViewTraffic.cpp.o' failed
make[2]: *** [src/gui/CMakeFiles/gui.dir/GUIViewTraffic.cpp.o] Error 1
CMakeFiles/Makefile2:4330: recipe for target 'src/gui/CMakeFiles/gui.dir/all' failed
make[1]: *** [src/gui/CMakeFiles/gui.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by behrisch** [comment]
Please make sure you also installed the correct devel package.
